### PR TITLE
Add OpType.CnRx and OpType.CnRz

### PR DESCRIPTION
**Pull request by sjdilkes**
# Related issues

[Please mention any github issues addressed by this PR.](https://github.com/CQCL/tket/issues/1346)

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Review by yao-cqc — APPROVED** [review]
Thanks, it's very thorough! Only a few minor points

**Review by yao-cqc — CHANGES_REQUESTED** [review]
hit approve by accident...

File: pytket/tests/circuit_test.py [hunk 1]
```diff
@@ -1536,6 +1536,20 @@ def test_pickle_bit() -> None:
...
+    c1rx.add_gate(OpType.CnRx, 0.3, [0, 1])
+    crx = Circuit(2)
+    crx.add_gate(OpType.CRx, 0.3, [0, 1])
+
+    c1rz = Circuit(2)
+    c1rz.add_gate(OpType.CnRz, 0.3, [0, 1])
+    crz = Circuit(2)
+    crz.add_gate(OpType.CRz, 0.3, [0, 1])
+
+    assert np.allclose(c1rz.get_unitary(), crz.get_unitary())
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply pytket/tests/circuit_test.py:1550]: oops, yep will add back in!

File: tket/include/tket/OpType/OpType.hpp [hunk 2]
```diff
@@ -608,6 +608,20 @@ enum class OpType {
...
 
+  /**
+   * Multiply-controlled \ref OpType::Rz
+   *
+   * The phase parameter is defined modulo \f$ 4 \f$.
+   */
+  CnRx,
+
+  /**
+   * Multiply-controlled \ref OpType::Rx
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply tket/include/tket/OpType/OpType.hpp:619]: done

**Pull request merged**